sources/packages/backend/libs/services/src/notifications/notification/notification.service.ts

```diff
@@ -70,57 +80,130 @@ export class NotificationService extends RecordDataModelService<Notification> {
   /**
    * Updates the date sent column of the inbox notification record.
    * @param notificationId notification id.
-   * @param entityManager optional repository that can be provided, for instance,
-   * to execute the command as part of an existing transaction. If not provided
-   * the local repository will be used instead.
+   * @param permanentFailureError Error details of a permanent failure if occurred while processing the notification.
    * @returns result of the record updated.
    */
   async updateNotification(
     notificationId: number,
-    entityManager?: EntityManager,
+    permanentFailureError?: PermanentFailureError[],
```

Comment:
`permanentFailureErrors`?

Reply:
The array object was created to match the error response type of GC Notify. But it is always treated as one permanent failure error.